daft/runners/ray_runner.py

```diff
@@ -619,6 +634,12 @@ def __init__(
                 max_task_backlog=max_task_backlog,
             )
```

Comment:
Do we need to call `stop_plan` here after the while loops finish to clean up?

Reply:
stop plan is called by the main thread to make this loop (in the background thread) complete and could be a race condition since the results may not be collected yet.